backend/municipal-services/ingestion-service/app/decorators/rbac_validator.py

```
@@ -0,0 +1,34 @@
+        for role_str in user_roles:
+            for user_role in UserRole:
+                if role_str["name"] == user_role.value:
+                    return request_info
+
+        raise HTTPException(
+            status_code=403,
+            detail=f"User role(s) '{user_roles}' not authorized for this operation. Allowed roles: {[role.value for role in UserRole]}",
+        )
+
+    except json.JSONDecodeError:
+        raise HTTPException(status_code=400, detail="Invalid RequestInfo format")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)